src/Raven.Server/Documents/PeriodicBackup/Retention/RetentionPolicyRunnerBase.cs

```diff
@@ -0,0 +1,145 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Raven.Client;
+using Raven.Client.Documents.Operations.Backups;
+using Raven.Server.Documents.PeriodicBackup.Restore;
+
+namespace Raven.Server.Documents.PeriodicBackup.Retention
+{
+    public abstract class RetentionPolicyRunnerBase
+    {
+        private readonly RetentionPolicy _retentionPolicy;
+        private readonly string _databaseName;
+
+        protected RetentionPolicyRunnerBase(RetentionPolicy retentionPolicy, string databaseName)
+        {
+            _retentionPolicy = retentionPolicy;
+            _databaseName = databaseName;
+        }
+
+        public abstract Task<List<string>> GetFolders();
+
+        public abstract Task<List<string>> GetFiles(string folder);
+
+        public abstract Task DeleteFolder(string folder);
+
+        public async Task Execute()
+        {
+            if (_retentionPolicy == null ||
+                _retentionPolicy.Disabled ||
+                _retentionPolicy.MinimumBackupsToKeep == null && _retentionPolicy.MinimumBackupAgeToKeep == null)
+                return; // no retention policy
+
+            if (_retentionPolicy.MinimumBackupsToKeep <= 0)
```

Comment:
Shouldn't be possible to set this to 0 or less, right? Are we checking that somewhere?

Reply:
Added validation of the configuration